src/components/atoms/UserAvatar/UserAvatar.scss

```diff
@@ -1,23 +1,34 @@
 @import "scss/constants";
 
 .UserAvatar {
-  $avatar-sizes-map: (
-    small: 25px,
-    medium: 40px,
-    large: 54px,
-    full: 100%,
-  );
-  $indicator-sizes-map: (
-    small: 8px,
-    medium: 10px,
-    large: 12px,
-    full: 25%,
-  );
-
-  // NOTE: parent and img mismatch due to legacy reasons
+  // @see https://math.stackexchange.com/questions/377234/circle-in-square-calculate-distance-from-squares-corner-to-circles-perimeter
+  $square-root-of-2: 1.41421356237;
+  $indicator-offset-factor: ($square-root-of-2 - 1) / 2;
+  $indicator-radius-factor: $square-root-of-2 / 2;
+
+  // indicator border should be 12% of the indicator size, but borders don't accept % so using px for default
+  $default-indicator-border-width: 2px;
+  $default-indicator-size: 28%;
   $default-avatar-size: 25px;
   $default-image-size: 100%;
-  $default-indicator-size: 8px;
+
+  // indicator border is 12% of indicator size which itself is some % of avatar size
+  // limited to min: 0.8px and max: 2px
+  @function indicator-border-width($indicator-size, $avatar-size) {
+    @return clamp(0.8px, 0.12 * #{$indicator-size} * #{$avatar-size}, 2px);
+  }
+
+  // order of elements:
+  // $modifier - one of UserAvatarSize
+  // $avatar-size - set in pixels or percent for larger than 54px
+  // $indicator-size - set in percent of parent
+  // $indicator-border-width - calculated as a percent of indicator size
+  $avatar-sizes-enum: (
+    (small, 25px, 28%, indicator-border-width(0.28, 25px)),
+    (medium, 40px, 28%, indicator-border-width(0.28, 40px)),
+    (large, 54px, 28%, indicator-border-width(0.28, 54px)),
+    (full, max(55px, 100%), 16%, indicator-border-width(0.16, 108px))
+  );
```

Comment:
This is basically an enumeration of related constants as per design spec.

The names `small`, `medium` and `large` are in line with pre-existing classes for `UserAvatar`, but if there's a need of the larger (than 54px) sizes to be codified as well, I can suggest moving to a Bootstrap-like nomenclature:

- Extra small (xs)
- Small (sm)
- Medium (md)
- Large (lg)
- Extra large (xl)
- Extra extra large (xxl)

Reply:
Good suggestion. This is aligned with our consts too. I don't think we need it now, but good to know for the future